src/videojs-hls.js

```diff
@@ -474,13 +474,20 @@ videojs.Hls.prototype.fillBuffer = function(offset) {
   }
 
   // resolve the segment URL relative to the playlist
+  segmentUri = this.resolveSegmentUrl(segment.uri);
+
+  this.loadSegment(segmentUri, offset);
+};
+
+videojs.Hls.prototype.resolveSegmentUrl = function(segmentRelativeUrl) {
```

Comment:
Yeah, me too.  How about:
1. resolvePlaylistRelativeUri
2. playlistUriToUrl
3. resolveUrl (might be hard to search for vs similar names)

Personally I like option 2 best, we are resolving a URI in the playlist (which may be a URL) to a URL.  PS, for people that always confuse the two (like me ;-)).   Here's a simple non-geekly explanation of the difference that is pretty accurate [URL vs URI](https://danielmiessler.com/study/url_vs_uri/)